tests/Integration/Behaviour/Features/Context/Domain/ContactFeatureContext.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use Behat\Gherkin\Node\TableNode;
+use PHPUnit_Framework_Assert;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Command\AddContactCommand;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Command\EditContactCommand;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Exception\ContactNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Contact\Query\GetContactForEditing;
+use PrestaShop\PrestaShop\Core\Domain\Contact\QueryResult\EditableContact;
+use PrestaShopBundle\Utils\BoolParser;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use function PrestaShopBundle\Utils\BoolParser;
+
+class ContactFeatureContext extends AbstractDomainFeatureContext
+{
+    private const DEFAULT_LOCALE_ID = 1; // EN locale
+    private const DUMMY_CONTACT_ID = 1;
+
+    /**
+     * @var int
+     */
+    private $defaultLangId;
+
+    public function __construct()
+    {
+        $configuration = CommonFeatureContext::getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->defaultLangId = $configuration->get('PS_LANG_DEFAULT');
+    }
+
+    /**
+     * @Given there is no contact with id :contactId
+     *
+     * @param int $contactId
+     */
+    public function thereIsNoContactWithId(int $contactId)
+    {
+        try {
+            $this->getQueryBus()->handle(new GetContactForEditing($contactId));
```

Comment:
this test will not fail if contact exists with given id.
You could use `\Tests\Integration\Behaviour\Features\Context\Util\NoExceptionAlthoughExpectedException` to make sure it fails when Contact is found

Reply:
Indeed considering the name of the method, it means you expect the `ContactNotFoundException` to be thrown so if you want to trigger this test then you can throw an exception right after the handle call
```suggestion
            $this->getQueryBus()->handle(new GetContactForEditing($contactId));
            throw new RuntimeException(sprintf('Contact found whereas not contact with id %s was expected', $contactId));
```